### PR TITLE
RGB LED Control V1.0

### PR DESCRIPTION
**Pull request by hazemashrafali**
Develop the GPIO Driver and use it to control RGB LED on the Tiva-C board using a push button
**quick links:**
[document record](https://drive.google.com/file/d/1rbtrGYNRV0gt-SiAdUYdJQ23SsswIVfZ/view?usp=sharing)
[hardware test](https://drive.google.com/file/d/1Z3dTkUQCXCWoM1CQG2R2u6zK3_ytbUS6/view?usp=sharing)
[design document](https://drive.google.com/file/d/1B69ezO3QPHtW2iSkeQZjQYSnctUBO0rU/view?usp=sharing)
[test sheet](https://drive.google.com/file/d/1ZIneQW9ridqKVOs4KlJeQCvewmK2dR7r/view?usp=sharing)

### CODE REVIEW

File: RGB LED Control V1.0/code/APP/app.c [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+		{
+			Delay_MS(50);
+			BUTTON_digitalRead(ENU_PORT_F,ENU_PIN_0,&button_value);
+			if(button_value == LOGIC_HIGH)
+			{
+				if(flag == FALSE)
+				{
+					press_count++;
+					switch (press_count)
+					{
```
Comment (ahmedatef23) [inline RGB LED Control V1.0/code/APP/app.c:56]: switch case should only have one break.

**Review by ahmedatef23 — APPROVED** [review]

File: RGB LED Control V1.0/code/APP/app.c (hunk 1 above)
Comment (hazemashrafali) [reply RGB LED Control V1.0/code/APP/app.c:56]: how to to use switch case with only one brake ?

Comment (ahmedatef23) [reply RGB LED Control V1.0/code/APP/app.c:56]: I think there are two solutions.
1. The easy way to replace the switch case with
2. The hard way is to use Fall-through switch cases.

Comment (hazemashrafali) [reply RGB LED Control V1.0/code/APP/app.c:56]: Done, you can check it now

Comment (hazemashrafali) [reply RGB LED Control V1.0/code/APP/app.c:56]: is there any refernce for the rule regard switch case shuld have only one break statement?

Comment (ahmedatef23) [reply RGB LED Control V1.0/code/APP/app.c:56]: The only available reference until now is our website

**Pull request merged**